File: java-example/src/main/java/com/pushbots/java_example/CustomHandler.java
package com.pushbots.java_example;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import com.pushbots.push.utils.PBConstants;

public class CustomHandler extends BroadcastReceiver {
    private static final String TAG = "CustomHandler";

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        Log.d(TAG, "action=" + action);

        // Handle Push Message when opened
        if (action.equals(PBConstants.EVENT_MSG_OPEN)) {

            //Bundle containing all fields of the opened notification
            Bundle bundle = intent.getExtras().getBundle(PBConstants.EVENT_MSG_OPEN);

            Log.i(TAG, "User clicked notification with Message: " + bundle.get("message"));

        } else if (action.equals(PBConstants.EVENT_MSG_RECEIVE)) {

            //Bundle containing all fields of the notification
            Bundle bundle = intent.getExtras().getBundle(PBConstants.EVENT_MSG_RECEIVE);
            Log.i(TAG, "User received notification with Message: " + bundle.get("message"));

        }

    }
}

File: kotlin-example/src/main/java/com/pushbots/example/CustomHandler.kt
package com.pushbots.example

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.pushbots.push.utils.PBConstants

class CustomHandler : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        Log.d(TAG, "action=$action")
        // Handle Push Message when opened
        if (action == PBConstants.EVENT_MSG_OPEN) { //Bundle containing all fields of the opened notification
            val bundle = intent.extras!!.getBundle(PBConstants.EVENT_MSG_OPEN)
            Log.i(
                TAG,
                "User clicked notification with Message: " + bundle!!["message"]
            )
        } else if (action == PBConstants.EVENT_MSG_RECEIVE) { //Bundle containing all fields of the notification
            val bundle = intent.extras!!.getBundle(PBConstants.EVENT_MSG_RECEIVE)
            Log.i(
                TAG,
                "User received notification with Message: " + bundle!!["message"]
            )
        }
    }

    companion object {
        private const val TAG = "CustomHandler"
    }
}